pkgs/misc/vscode-extensions/python/default.nix

```diff
@@ -14,6 +20,40 @@ assert ctagsUseFixed -> null != ctags;
 let
   pythonDefaultsTo = if pythonUseFixed then "${python}/bin/python" else "python";
   ctagsDefaultsTo = if ctagsUseFixed then "${ctags}/bin/ctags" else "ctags";
+
+  # The arch tag comes from 'PlatformName' defined here:
+  # https://github.com/Microsoft/vscode-python/blob/master/src/client/activation/types.ts
+  archTag =
+    if stdenv.isLinux && stdenv.isx86_64 then "linux-x64"
+    else if stdenv.isDarwin then "osx-x64"
+    else throw "Only x86_64 Linux and Darwin are supported.";
+
+  extractNuGet = { name, version, src, ... }:
```

Comment:
Why not use https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/fetchnuget/default.nix?